CRM/Contribute/Form/AbstractEditPayment.php

```diff
@@ -432,12 +432,14 @@ public function assignProcessors() {
       }
       $this->_processors = array();
       foreach ($this->_paymentProcessors as $id => $processor) {
-        $this->_processors[$id] = ts($processor['name']);
-        if (!empty($processor['description'])) {
-          $this->_processors[$id] .= ' : ' . ts($processor['description']);
-        }
-        if ($processor['is_recur']) {
-          $this->_recurPaymentProcessors[$id] = $this->_processors[$id];
+        if ($processor['is_test'] == ($this->_mode == 'test')) {
```

Comment:
"when you want to submit a payment in test mode you use the id of the live processor, so maybe we just always need those?" - I've been trying to ditch that & make it use the ACTUAL processor id - test or live